k8s/upgrades/0.8.2-0.9.0/jiva/jiva_upgrade_runtask.yaml

```diff
@@ -329,6 +316,82 @@ spec:
 kind: RunTask
+metadata:
+  name: upgrade-jiva-volume-0.8.2-0.9.0-pre-check-ctrl-pod-phase
+  namespace: default
+spec:
+  meta: |
+    id: listCtrlPods
+    runNamespace: {{ .UpgradeItem.namespace }}
+    apiVersion: v1
+    kind: Pod
+    action: list
+    options: |-
+      labelSelector: openebs.io/persistent-volume={{ .UpgradeItem.name }},openebs.io/controller=jiva-controller
+  post: |
+    {{- $CustomJsonPath := printf "{.items[?(@.status.phase=='Running')].metadata.name}" -}}
+
+    {{- jsonpath .JsonResult $CustomJsonPath | trim | saveAs "listCtrlPods.podName" .TaskResult | noop -}}
+    {{- .TaskResult.listCtrlPods.podName | notFoundErr "controller pod is not in running phase" | saveIf "listCtrlPods.notFoundErr" .TaskResult | noop -}}
+
+    {{- .TaskResult.listCtrlPods.podName | default "" | splitList " " | len | saveAs "listCtrlPods.actualRunningPodCount" .TaskResult -}}
+
+    {{- $expectedPodCount := .TaskResult.listTargetDeployment.replicaCount | int -}}
+    {{- $msg := printf "expected %v no of running replica pod(s), found only %v replica pod(s)" $expectedPodCount .TaskResult.listCtrlPods.actualRunningPodCount -}}
+    {{- .TaskResult.listCtrlPods.nodeNames | default "" | splitList " " | isLen $expectedPodCount | not | verifyErr $msg | saveIf "listCtrlPods.verifyErr" .TaskResult | noop -}}
+
+    {{- $message := printf "jiva controller pods are in running phase for volume: {%s}" .UpgradeItem.name -}}
+    {{- $status := .Config.successStatus.value -}}
+
+    {{- $URName := upgradeResultWithTaskOwnerName .UpgradeItem.upgradeResultName -}}
+    {{- $URNamespace := upgradeResultWithTaskOwnerNamespace .UpgradeItem.upgradeResultNamespace -}}
+    {{- $taskName := upgradeResultWithTaskName "upgrade-jiva-volume-0.8.2-0.9.0-pre-check-ctrl-pod-phase" -}}
+    {{- $taskStatus := upgradeResultWithTaskStatus $status -}}
+    {{- $taskMessage := upgradeResultWithTaskMessage $message -}}
+    {{- $taskStartTime := upgradeResultWithTaskStartTime now -}}
+    {{- $taskEndTime := upgradeResultWithTaskEndTime now -}}
+    {{- upgradeResultUpdateTasks $taskStartTime $URName $URNamespace $taskName $taskStatus $taskMessage $taskEndTime -}}
+---
+apiVersion: openebs.io/v1alpha1
+kind: RunTask
+metadata:
+  name: upgrade-jiva-volume-0.8.2-0.9.0-pre-check-replica-pod-phase
+  namespace: default
+spec:
+  meta: |
+    id: listReplicaPods
+    runNamespace: {{ .UpgradeItem.namespace }}
+    apiVersion: v1
+    kind: Pod
+    action: list
+    options: |-
+      labelSelector: openebs.io/persistent-volume={{ .UpgradeItem.name }},openebs.io/replica=jiva-replica
+  post: |
+    {{- $CustomJsonPath := printf "{.items[?(@.status.phase=='Running')].spec.nodeName}" -}}
+
+    {{- jsonpath .JsonResult $CustomJsonPath | trim | saveAs "listReplicaPods.nodeNames" .TaskResult | noop -}}
+    {{- .TaskResult.listReplicaPods.nodeNames | notFoundErr "replica pods nodeNames are not found" | saveIf "listReplicaPods.notFoundErr" .TaskResult | noop -}}
```

Comment:
Message can be `No nodes found for jiva replica pods`

Reply:
Done